lib/src/sign_tool.dart

```diff
@@ -15,44 +15,138 @@ class SignTool {
   final Logger _logger = GetIt.I<Logger>();
   final Configuration _config = GetIt.I<Configuration>();
 
-  /// Use Powershell script to get the Publisher ("Subject") of the certificate
+  /// get the certificate "Subject" for the Publisher value
   Future<void> getCertificatePublisher() async {
     _logger.trace('getting certificate publisher');
 
+    var subject = '';
+
+    if (_config.signToolOptions != null) {
+      if (_config.signToolOptions!.containsArgument('/sha1')) {
+        subject = await _getCertificateSubjectByThumbprint();
+      } else if (_config.signToolOptions!.containsArguments(['/n', '/r'])) {
+        subject = await _getCertificateSubjectBySubject();
+      } else if (_config.signToolOptions!.containsArgument('/i')) {
+        subject = await _getCertificateSubjectByIssuer();
+      }
+    } else if (_config.certificatePath != null &&
+        extension(_config.certificatePath!).toLowerCase() == '.pfx') {
+      subject = await _getPfxCertificateSubject();
+    }
```

Comment:
Quick dry code - have not tested it....
```
else if (_config.certificatePath != null)  {
  subject = await _getCrtCertificateSubject();
}
```


```
  Future<String> _getCrtCertificateSubject() async {
    _logger.trace('getting crt certificate Subject');

    var certificateDetailsProcess = await _executePowershellCommand(
        "new-object System.Security.Cryptography.X509Certificates.X509Certificate2(\"${_config.certificatePath}\") | select -expandproperty Subject -First 1
");

    certificateDetailsProcess.exitOnError();

    var subject = (certificateDetailsProcess.stdout as String).trim();

    return subject;
  }
```

Reply:
done here: https://github.com/YehudaKremer/msix/pull/156/commits/b252a98818abbd2b2341525aa283c2e9e02af00b

Reply:
also added the documentation https://github.com/YehudaKremer/msix/pull/156/commits/bc0d17ef64e3d35e9018a0d085e23b9ad9368974